modules/react-components/src/components/header/header.tsx

```diff
@@ -385,7 +385,22 @@ export const Header: FunctionComponent<HeaderPropsInterface> = (
             return null;
         }
 
-        return userDropdownLinks.map((category: HeaderLinkCategoryInterface, categoryIndex: number) => {
+        const adjustedUserDropdownLinks : HeaderLinkCategoryInterface[] = userDropdownLinks
+            .reduce((previousValue, currentValue) => {
```

Comment:
Let's refactor in a followup PR.
```suggestion
            .reduce((previous, current) => {
```

Reply:
Fixed with https://github.com/wso2/identity-apps/pull/2893